files/build_templates/qos_config.j2

```diff
@@ -27,6 +27,13 @@
 {%- endfor %}
 {%- set port_names_active = port_names_list_active | join(',') -%}
 
+{%- set port_names_list_extra_queues = [] %}
+{%- for port in PORT_ACTIVE %}
+    {% if (generate_dscp_to_tc_map is defined) and ('type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] == 'LeafRouter' and DEVICE_NEIGHBOR is defined and 'name' in DEVICE_NEIGHBOR[port] and 'T0' in DEVICE_NEIGHBOR[port]['name']) or
```

Comment:
Do not use the name to derive the neighbor role. That could be misleading. use the 'type' field in the DEVICE_NEIGHBOR_METADATA. Refer https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/buffers_config.j2#L47 for the usage